src/librustc/hir/lowering.rs

```diff
@@ -174,15 +174,16 @@ enum ImplTraitContext<'a> {
     /// Example: `fn foo(x: impl Debug)`, where `impl Debug` is conceptually
     /// equivalent to a fresh universal parameter like `fn foo<T: Debug>(x: T)`.
     ///
-    /// Newly generated parameters should be inserted into the given `Vec`
+    /// Newly generated parameters should be inserted into the given `Vec`.
     Universal(&'a mut Vec<hir::GenericParam>),
 
-    /// Treat `impl Trait` as shorthand for a new universal existential parameter.
+    /// Treat `impl Trait` as shorthand for a new existential parameter.
     /// Example: `fn foo() -> impl Debug`, where `impl Debug` is conceptually
-    /// equivalent to a fresh existential parameter like `abstract type T; fn foo() -> T`.
+    /// equivalent to a fresh existential parameter like `existential type T; fn foo() -> T`.
     ///
-    /// We store a DefId here so we can look up necessary information later
-    Existential(DefId),
+    /// We store a `DefId` for the parent item here so we can look up necessary information
```

Comment:
Can you add a comment about what it means when the `DefId` is `None`?

Reply:
Sure.